src/tasks/WasmAppBuilder/InterpToNativeGenerator.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using System.Diagnostics.CodeAnalysis;
+
+//
+// This class generates the icall_trampoline_dispatch () function used by the interpreter to call native code on WASM.
+// It should be kept in sync with mono_wasm_interp_to_native_trampoline () in the runtime.
+//
+
+#nullable enable
+
+public class InterpToNativeGenerator
+{
+    protected TaskLoggingHelper Log { get; set; }
+
+    public InterpToNativeGenerator(TaskLoggingHelper log) => Log = log;
+
+    private static string TypeToSigType(char c) => c switch
+    {
+        'V' => "void",
+        'I' => "int",
+        'L' => "int64_t",
+        'F' => "float",
+        'D' => "double",
+        _ => throw new Exception("Can't handle " + c)
```

Comment:
This should indicate a bug in the task, correct? It should be eliminated though if we use an enum string.
In any case, if we want to throw an exception here, then use a more specific type which can be caught by the caller, and a useful error message with details of the method causing this be emitted.

Reply:
Yes, it indicates bug inside of the task. Other generators produce signature that this one can't handle. Part of the working with this string signature is also in the c code and here it is used only internally. I think that converting it to a list of enums isn't necessary. Anyway I have refactored so most of the work is handled in `SignatureMapper`.